### ISSUE TITLE
What should routines giving file/module/function name and line number be called

### ISSUE THREAD

**Issue by daviditen** [chapel-lang/chapel#13680]
In master there are now routines to return line number, filename, module name, and function name.  They are currently named:
`Reflection.lineNumber`,
`Reflection.fileName`,
`Reflection.moduleName`,
`Reflection.functionName`.

Should these be renamed? If yes, what should they be called?

**Comment by bradcray** [comment]
Speaking for myself, `fileName` strikes me as one of those "should really be considered one-word" terms that I'd personally capitalize as `filename`.  But then that gets into a question about whether that's too common of an identifier to utilize in this way in a standard module (albeit one that I don't think is `use`d by default and so not something that is likely to bite programmers, particularly as we do more and more `private use` by default conversions?)

I've also become increasingly soured on referring to "procedures" and "iterators" collectively as "functions" in Chapel (though I realize we have tons of documentation and text that does) because they don't really match the mathematical notion and are not functional by the PL terminology.  Mentally, I find myself preferring "subroutine" or simply "routine" lately...  That said, it could be that this ship has long since sailed for us and isn't worth reconsidering.

I also wonder a little bit whether using paren-less identifiers is the right approach for these.  I think in the past we've used paren-less methods/procedures when the thing being computed could just as easily be stored in a variable as computed on the fly.  But these don't behave in a variable-like manner given that code like:

```chapel
writeln(lineNumber);
writeln(lineNumber);
writeln(lineNumber);
```

seems like it should print the same thing three times if `lineNumber` were just a variable, but won't.

**Comment by daviditen** [comment]
Taking the suggestions so far, we'd have:

`Reflection.getLineNumber()`,
`Reflection.getFilename()`,
`Reflection.getModuleName()`,
`Reflection.getRoutineName()`

These names would be OK with me.

**Comment by dlongnecke-cray** [comment]
I have a personal preference for

`getFileName` `getModuleName` `getRoutineName`

But not enough of one to fight to the death over it.

**Comment by vasslitvinov** [comment]
While `function` in this context is not ideal, I would not like `routine` either - because it is more of a mouthful and because of Fortran connotations. The latter is hardly a consideration for Python converts, though. :)

We struggled with this a while back when deciding how to organize the spec. I am not seeing any arguments today that we did not have then, whatever they were. So one way out is to uphold that decision. Consistency of documentation is more important to me than finding a perfect name.

I have a strong, albeit non-violent, preference for considering "filename" as two words. Considering it oneword feels to me like jargon, as I personally enjoy `filename` and `linenum` in our compiler without uppercase characters.

I vote for `moduleName()` style.

Because the `get` part is valuable to me only when there is a corresponding `put` or `set`.
Paren-less version would look too much like a variable if I write `Reflection.moduleName` .
